### PR TITLE
Migrate to Java 21

### PR DESCRIPTION
**Pull request by HindujaB**
## Purpose
$subject

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java [hunk 1]
```diff
@@ -374,53 +372,45 @@ public static void writeByteStreamToOutputStream(Environment env, BObject entity
...
-                    throw ErrorCreator.createError(StringUtils.fromString(
-                            "Error occurred while writing the stream content: "
-                                    + MimeUtil.removeJavaExceptionPrefix(e.getMessage())));
-                }
-                writeContent(env, entity, outputStream, iteratorObj, latch);
-            }
+                                     BObject iteratorObj) {
+        try {
+            Object result = env.getRuntime().callMethod(iteratorObj, BYTE_STREAM_NEXT_FUNC,
+                    new StrandMetadata(true, null));
```
Comment (TharmiganK) [inline native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java:383]: Here, we are reading content from a ballerina stream and write it back to a java output stream. My understanding is that `new StrandMetadata(true, null)` creates a new strand. In that case should we create a new `StrandMetadata` for each call of the `writeContent` method?

File: native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java [hunk 2]
```diff
@@ -442,31 +432,35 @@ public static void writeEventStreamToOutputStream(Environment env, BObject entit
...
-                    @Override
-                    public void notifyFailure(BError bError) {
-                        OutputStream outputStream = (OutputStream) eventStreamWriter.getNativeData(OUTPUT_STREAM);
-                        EntityBodyHandler.closeMessageOutputStream(outputStream);
-                        throw ErrorCreator.createError(StringUtils.fromString(
-                                "Error occurred while streaming content: " + bError.getMessage()));
-                    }
-                }, null, null);
+        try {
+            handleEventPanic(eventStreamWriter, env.getRuntime().callMethod(eventStreamWriter,
```
Comment (TharmiganK) [inline native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java:436]: Why this function is named as `handleEventPanic`? Isn't this a success scenario?

Comment (HindujaB) [reply native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java:436]: this seems like a bug. I will update it

File: native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java (hunk 1 above)
Comment (warunalakshitha) [reply native/src/main/java/io/ballerina/stdlib/mime/util/EntityBodyHandler.java:383]: Yes. We do not need to pass strand metadata here.

**Review by TharmiganK — DISMISSED** [review]
LGTM

**Review by TharmiganK — APPROVED** [review]

**Review by warunalakshitha — APPROVED** [review]

**Pull request merged**